### PR TITLE
Add community_emoji to SiteConfig

### PR DESCRIPTION
**Pull request by vaidehijoshi**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

As mentioned in https://github.com/forem/forem/issues/11449, we currently hardcode some emojis in the `<title>` tags of our site, but that locks in other Forem creators to the hardcoded emojis that we chose 😕 

Instead, we want to allow Forem creators to choose _their own_ emojis to associate with the site (or to not have an emoji at all!). This PR removes the hardcoded emojis and add an optional `community_emoji` field to `SiteConfig`, which can be blank or which a Forem creator can override with their own choice of emoji.

It also removes a hardcoded emoji on the `/connect` page, since associating an emoji with a specific _page_ is a much larger feature and seems like it deserves a product discussion.

## Related Tickets & Documents

Closes https://github.com/forem/forem/issues/11449.

## QA Instructions, Screenshots, Recordings

Starting with a fresh Forem/SiteConfig:
1. Head to `/admin/config`. You should see the default emoji in the `community_emoji` field within `Community Content`:
one
2. Try clearing out the emoji entirely -- you should be able to do it with no problems since it is _not_ a required field:
two
3. Try adding some 🥐s for emojis. You should be able to update the SiteConfig without any issues:
three
4. Check the `<title>` tag on the articles index page. You should see your community emojis reflected in the DOM!

## Added tests?

- [x] Yes (added a new test and updated preexisting tests!)
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## Are there any post deployment tasks we need to perform?

**We will need to add `👩‍💻👨‍💻` as the `community_emoji` on DEV. We should also add them to forem.team and forem.dev if we want something _other than_ the default `🌱`.**

## What ~gif~ EMOJI best describes this PR or how it makes you feel?
🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐🥐

### CODE REVIEW

File: app/views/chat_channels/index.html.erb
Comment (vaidehijoshi) [inline app/views/chat_channels/index.html.erb:1]: This is the hardcoded emoji I removed, as mentioned in the PR description.

**Comment by pkfrank** [comment]
Do we think there are any SEO implications that we should be aware of on DEV?  I know that we intend to reinstate the emojis  on `dev.to` once this is merged in but just want to make sure we've given that a quick thought.

**Review by mstruve — APPROVED** [review]
LOVE IT! I think the CodeClimate about instance variables in this case since it is site wide and not an issue with threading is safe to ignore.

**Comment by vaidehijoshi** [comment]
> Do we think there are any SEO implications that we should be aware of on DEV? I know that we intend to reinstate the emojis on `dev.to` once this is merged in but just want to make sure we've given that a quick thought.

I _do_ think it will affect SEO (at least until we add the 2 emojis we have hardcoded back into our own SiteConfig) but I am not sure what our general "best practices" around this have been. I took a look at [a previous PR](https://github.com/forem/forem/pull/6887) where we made these kinds of "generalization" changes and I didn't see any flags that made it evident that we can't change any `<title>` tags...but I agree with you that we should double check that.

I wonder if @benhalpern has insights here about whether it's safe to proceed/what we should do if it _isn't_.

**Review by Zhao-Andy — APPROVED** [review]
Nice!! Yay to emojis!

Re: SEO. If we think it's a big enough concern then we could split the SiteConfig addition with the title change addition. Personally think it's more trouble than it's worth, since we would probably have only a minute or two where the emojis were missing from the dev.to `<title>`.

**Comment by pkfrank** [comment]
Can we consider this "do not merge" until @benhalpern weighs in?

It's probably fine, but minimal reason to _not_ treat this with some extra care as these SEO things can be fickle.

**Comment by vaidehijoshi** [comment]
@pkfrank sure thing—have converted this to a draft until we get @benhalpern’s weigh in here 👍🏽

**Comment by vaidehijoshi** [comment]
I slept on this and had some new thoughts on how to proceed here:

Maybe the right thing to do is just to remove the hardcoded emojis completely and not even add a `community_emoji` field at all. If we add it, we're now basically affording a Forem admin a way of changing this emoji an indefinite amount of times, which will impact the `<title>` for that Forem instance. The side effect of that is that we're allowing them to mess around with their SEO, potentially, which we might not want to encourage by adding the emoji as a configurable field to `SiteConfig`.

The other issue here is that I'm not currently validating that the field itself is an emoji -- I could add this, but validating emojis in a regex is not necessarily simple from what I can tell, and I would like to avoid [adding any dependencies](https://github.com/janlelis/unicode-emoji) for this.

These ^^ reasons, alongside the general weariness of [adding yet another value to SiteConfig](https://github.com/forem/forem/issues/11449#issuecomment-728814375) makes me think that this "feature" is actually maybe not worth it. I kind of feel like we should remove the hardcoded emojis from the codebase and call it a day.

Curious what folks + @benhalpern think about this (also curious re: his understanding of SEO consequences)

**Comment by benhalpern** [comment]
I like this. We already don't use emojis in key leaf pages for the SEO paranoia factor, so unless this changes that behavior (and at a glance I don't see that it does) this seems cool.

If we ever move away from having emojis as part of the title, we can ditch this feature and alert admins, but otherwise there might be new places we could include these (including places like forem.com where it might be fun to reference communities by their emojis)

TLDR... Yay! (I also marked the codeclimate as wontfix)

**Comment by vaidehijoshi** [comment]
UPDATE: I've confirmed with @benhalpern that this approach is okay for SEO, which also makes me comfortable with rolling this out to Forem creators/admins as well, since I know it's not going to destroy their SEO 😅 

However, I _have_ added some validations to ensure that non-emoji characters can't be used in this field. Apparently we already had the `emoji_regex` gem as a dependency, so I was able to use it here!! 🙌 

**TO QA MY RECENT UPDATES, PLEASE DO THE FOLLOWING:**
1) Try updating the `community_emoji` field with a non-emoji:
Screen_Shot_2020-11-17_at_11_53_02_AM
...it should not update and you should get an error message:
Screen Shot 2020-11-17 at 11 53 28 AM

2) Try updating the field with any valid emoji -- you shouldn't see any errors, and the field should update:
Screen_Shot_2020-11-17_at_11_54_21_AM

3) Try setting the `community_emoji` to blank -- you shouldn't see any errors, and the field should update:
Screen_Shot_2020-11-17_at_11_55_12_AM

Am re-requesting @mstruve + @Zhao-Andy's reviews here as I've added some changes since their approval(s)

**Review by Zhao-Andy — APPROVED** [review]
👍 👍 👍

File: spec/requests/admin/configs_spec.rb
Comment (mstruve) [inline spec/requests/admin/configs_spec.rb:137]: +1

**Review by mstruve — APPROVED** [review]
TIL alongside is one word. Also, the code looks pretty sound as well 😉  🚀

**Pull request merged**
